mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java

```diff
@@ -186,42 +183,19 @@ void registerGroupShouldCreateRetryExchange() throws Exception {
     }
 
     @Test
-    void rabbitMQChannelPoolDoesntLimitNumberOfChannelsCreated() {
+    void rabbitMQChannelPoolShouldThrowWhenTimeoutAndAllChannelsAreTaken() {
```

Comment:
Why is there a timeout? Aren't we willing to wait?

BTW isn't our problem a too big number of "publisher" threads?

Reply:
I let the client waits, when exceed timeout then I reject the client. In case heavy load with long operations, of all channel are acquired, then if I let clients wait forever then it may leads to the situation of blocking elastic scheduler, make james consume more resources to do nothing.

> BTW isn't our problem a too big number of "publisher" threads

I can set limitRate() for dispatching operations. But IDK which number is valid